### PR TITLE
i18n (hi): update `nav.ts` for hindi translated content

### PR DESCRIPTION
**Pull request by DevRohit06**
Description (required)
This PR performs a translation change in the documentation navigation bar

Suggested label: i18n

### CODE REVIEW

File: src/i18n/hi/nav.ts [hunk 1]
```diff
@@ -1,11 +1,74 @@
 import { NavDictionary } from '../translation-checkers';
 
 export default NavDictionary({
-	// Start Here
 	startHere: 'यहाँ से शुरू करे',
 	'getting-started': 'पहले कदम',
+	install: 'स्थापित करें',
+	'editor-setup': 'संपादक सेटअप',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:7]: `संपादक` here sounds more like a person (ex: newspaper editor). I find `एडिटर` more suitable in this case
```suggestion
	'editor-setup': 'एडिटर सेटअप',
```

Comment (DevRohit06) [reply src/i18n/hi/nav.ts]: so should we leave the words which sounds complicated or uncommon?

File: src/i18n/hi/nav.ts [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
+	'guides/upgrade-to/v3': 'v3 में अपग्रेड करें',
 
-	// Core Concepts
-	coreConcepts: 'मूल अवधारणाएं',
-	'concepts/why-astro': 'क्यों Astro',
+	coreConcepts: 'मूल अवधारणाएँ',
+	'concepts/why-astro': 'Astro क्यों',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:11]: This is admissible but with this, the [hi/why-astro](https://github.com/withastro/docs/blob/main/src/content/docs/hi/concepts/why-astro.mdx#L2) title tag should also be updated accordingly (nothing urgent but needs to addressed)

**Comment by at-the-vr** [comment]
Hey @DevRohit06 love the energy with your commits. As for the `संपादक` change, it was more of a subjective suggestion rather than the word being complex or uncommon.

**Comment by DevRohit06** [comment]
> Hey @DevRohit06 love the energy with your commits. As for the `संपादक` change, it was more of a subjective suggestion rather than the word being complex or uncommon.

I understood, and I'll keep it in mind next time when I'll do the pull request.

File: src/i18n/hi/nav.ts [hunk 3]
```diff
@@ -1,11 +1,74 @@
...
+	'blog-tutorial': 'ब्लॉग बनाएं',
+	'add-collections-tutorial': 'सामग्री संग्रह बढ़ाएं',
+	'add-transitions-tutorial': 'दृश्य परिसर बढ़ाएं',
+
+	basics: 'मौलिक अवधारणाएँ',
+	'core-concepts/project-structure': 'परियोजना संरचना',
+	'core-concepts/astro-components': 'घटक',
+	'core-concepts/astro-pages': 'पृष्ठ',
+	'core-concepts/layouts': 'नक़्शे',
+	'core-concepts/astro-syntax': 'Astro की सिंटैक्स',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:24]: I believe `अवयव` is better alternative to use (I have the same word running on the translated pages for Astro Docs), also Rendering Modes is one of the newer pages merged from PR #4667 
```suggestion
	basics: 'मौलिक अवधारणाएँ',
	'core-concepts/project-structure': 'परियोजना संरचना',
	'core-concepts/astro-components': 'अवयव',
	'core-concepts/astro-pages': 'पृष्ठ',
	'core-concepts/layouts': 'नक़्शे',
	'core-concepts/astro-syntax': 'Astro की सिंटैक्स',
	'core-concepts/rendering-modes': 'अनुवाद मध्यम'
```

File: src/i18n/hi/nav.ts [hunk 4]
```diff
@@ -1,11 +1,74 @@
...
+	'guides/view-transitions': 'view transitions',
+
+	addons: 'एड-ऑन्स',
+	'guides/integrations-guide': 'एकीकरण जानकारी',
+	'core-concepts/framework-components': 'UI फ़्रेमवर्क्स',
+	'guides/server-side-rendering': 'सर्वर-साइड रेंडरिंग (SSR)',
+
+	examples: 'उदाहरण',
+	'guides/migrate-to-astro': 'Astro में माइग्रेट करें',
+	'guides/cms': 'एक CMS से कनेक्ट करें',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:37]:
```suggestion
	'guides/cms': 'एक CMS से जुड़े ',
```

File: src/i18n/hi/nav.ts [hunk 5]
```diff
@@ -1,11 +1,74 @@
...
+	'guides/styling': 'स्टाइल और CSS',
+	'guides/images': 'छवियाँ',
+	'guides/fonts': 'फ़ॉन्ट्स',
+	'guides/imports': 'आयात',
+	'core-concepts/endpoints': 'एंडपॉइंट्स',
+	'guides/data-fetching': 'डेटा फ़ेचिंग',
+	'guides/internationalization': 'अंतर्राष्ट्रीयकरण',
+	'guides/middleware': 'मिडलवेयर',
+	'guides/testing': 'टेस्टिंग',
+	'guides/troubleshooting': 'समस्या समाधान',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:55]: Technologies should be kept as it is (ex: React, Astro) as a general rule
```suggestion
	features: 'गाइड्स',
	'core-concepts/routing': 'रूटिंग',
	'guides/markdown-content': 'Markdown और MDX',
	'guides/client-side-scripts': 'स्क्रिप्ट और घटना संचालन',
	'guides/styling': 'स्टाइल और CSS',
	'guides/images': 'छवियाँ',
	'guides/fonts': 'फ़ॉन्ट्स',
	'guides/imports': 'आयात',
	'core-concepts/endpoints': 'एंडपॉइंट्स',
	'guides/data-fetching': 'डेटा फ़ेचिंग',
	'guides/internationalization': 'अंतर्राष्ट्रीयकरण',
	'guides/middleware': 'मिडलवेयर',
	'guides/testing': 'परिक्षण',
	'guides/troubleshooting': 'समस्या समाधान',
```

File: src/i18n/hi/nav.ts [hunk 6]
```diff
@@ -1,11 +1,74 @@
...
+	'reference/configuration-reference': 'कॉन्फ़िगरेशन',
+	'reference/api-reference': 'रनटाइम API',
+	'reference/integrations-reference': 'एकीकरण API',
+	'reference/adapter-reference': 'एडाप्टर API',
+	'reference/image-service-reference': 'इमेज सेवा API',
+	'reference/dev-overlay-plugin-reference': 'डेवलपर ओवरले प्लगइन API',
+	'reference/directives-reference': 'टेम्पलेट निर्देशिकाएँ',
+	'reference/cli-reference': 'Astro CLI',
+	'reference/error-reference': 'त्रुटि संदर्भ',
+	'guides/publish-to-npm': 'NPM पैकेज प्रारूप',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:73]:
```suggestion
	reference: 'संदर्भ',
	'reference/configuration-reference': 'कॉन्फ़िगरेशन',
	'reference/api-reference': 'रनटाइम API',
	'reference/integrations-reference': 'एकीकरण API',
	'reference/adapter-reference': 'एडाप्टर API',
	'reference/image-service-reference': 'छवि सेवा API',
	'reference/dev-overlay-plugin-reference': 'डेवलपर ओवरले प्लगइन API',
	'reference/directives-reference': 'टेम्पलेट निर्देशिकाएँ',
	'reference/cli-reference': 'Astro CLI',
	'reference/error-reference': 'त्रुटि संदर्भ',
	'guides/publish-to-npm': 'NPM पैकेज प्रारूप',
```

File: src/i18n/hi/nav.ts [hunk 7]
```diff
@@ -1,11 +1,74 @@
...
+	basics: 'मौलिक अवधारणाएँ',
+	'core-concepts/project-structure': 'परियोजना संरचना',
+	'core-concepts/astro-components': 'घटक',
+	'core-concepts/astro-pages': 'पृष्ठ',
+	'core-concepts/layouts': 'नक़्शे',
+	'core-concepts/astro-syntax': 'Astro की सिंटैक्स',
+
+	builtins: 'सामग्री स्थापित',
+	'guides/content-collections': 'सामग्री संग्रह',
+	'guides/view-transitions': 'view transitions',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:29]: Another page with Astro Release 3.5 [Prefetch](https://astro.build/blog/astro-350/#prefetch)
```suggestion
	builtins: 'बिल्ट-इंस ',
	'guides/content-collections': 'सामग्री संग्रह',
	'guides/view-transitions': 'व्यू-ट्रांज़िशन्स ',
	'guides/prefetch': 'प्रीफ़ेच',
```

File: src/i18n/hi/nav.ts [hunk 8]
```diff
@@ -1,11 +1,74 @@
...
+	'core-concepts/endpoints': 'एंडपॉइंट्स',
+	'guides/data-fetching': 'डेटा फ़ेचिंग',
+	'guides/internationalization': 'अंतर्राष्ट्रीयकरण',
+	'guides/middleware': 'मिडलवेयर',
+	'guides/testing': 'टेस्टिंग',
+	'guides/troubleshooting': 'समस्या समाधान',
+
+	configuration: 'कॉन्फ़िगरेशन',
+	'guides/configuring-astro': 'Astro कॉन्फ़िगरेशन फ़ाइल',
+	'guides/typescript': 'टाइपस्क्रिप्ट',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:61]:
```suggestion
	'guides/typescript': 'Typescript',
```

File: src/i18n/hi/nav.ts [hunk 9]
```diff
@@ -1,11 +1,76 @@
...
-	// Core Concepts
-	coreConcepts: 'मूल अवधारणाएं',
+	coreConcepts: 'मूल अवधारणाएँ',
 	'concepts/why-astro': 'क्यों Astro',
+	'concepts/islands': 'Astro द्वीप',
+
+	tutorials: 'शिक्षण सत्र',
+	'blog-tutorial': 'ब्लॉग बनाएं',
+	'add-collections-tutorial': 'सामग्री संग्रह बढ़ाएं',
+	'add-transitions-tutorial': 'दृश्य परिसर बढ़ाएं',
```
Comment (at-the-vr) [inline src/i18n/hi/nav.ts:17]:
```suggestion
	'add-transitions-tutorial': 'व्यू-ट्रांज़िशन्स बढ़ाएं',
```

**Review by at-the-vr — APPROVED** [review]
Amazing Response time, LGTM

**Comment by DevRohit06** [comment]
> Amazing Response time, LGTM

😅😅 Thanks

**Review by Yan-Thomas — APPROVED** [review]
Woah, amazing work! Thanks peeps 🚀

**Pull request merged**